src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -160,6 +161,18 @@ public class Session {
             HostToClientWorker(_process.StandardOutput.BaseStream, hostEnd).DoNotWait();
         }
 
+        private void StartSession() {
+            _process.Start();
+            _process.WaitForExit(250);
+            if (_process.HasExited && _process.ExitCode != 0) {
+                var message = ErrorCodeConverter.MessageFromErrorCode(_process.ExitCode);
```

Comment:
`Process.ExitCode` is the errorlevel of the process, e.g. what you pass to `exit(...)`, and meaning of those is defined by the process itself. For R host, the user can cause it to return an arbitrary errorlevel - `quit(status = ...)`. If we always try to translate that to a Win32 error, we'll be generating spurious messages.